apps/src/templates/sectionProgress/detail/VirtualizedDetailView.jsx

```diff
@@ -71,26 +74,34 @@ class VirtualizedDetailView extends Component {
   static propTypes = {
     section: sectionDataPropType.isRequired,
     scriptData: scriptDataPropType.isRequired,
+    levelProgressByStudent: PropTypes.objectOf(
+      PropTypes.objectOf(studentLevelProgressType)
+    ).isRequired,
     lessonOfInterest: PropTypes.number.isRequired,
     setLessonOfInterest: PropTypes.func.isRequired,
     columnWidths: PropTypes.arrayOf(PropTypes.number).isRequired,
-    levelsByLesson: PropTypes.object,
     onScroll: PropTypes.func,
-    stageExtrasEnabled: PropTypes.bool
+    scriptName: PropTypes.string,
+    scriptId: PropTypes.number
   };
 
   constructor(props) {
     super(props);
-    this.state = {
-      fixedColumnCount: 1,
-      fixedRowCount: 2,
-      scrollToColumn: 0,
-      scrollToRow: 0
-    };
-    this.detailView = null;
     this.setDetailViewRef = this.setDetailViewRef.bind(this);
+    this.onClickLevel = this.onClickLevel.bind(this);
+    this.cellRenderer = this.cellRenderer.bind(this);
+    this.studentCellRenderer = this.studentCellRenderer.bind(this);
+    this.getColumnWidth = this.getColumnWidth.bind(this);
   }
 
+  state = {
```

Comment:
our style guide isn't opinionated about this so in some style guide update proposals i'm working on i want to propose that we declare all class properties that don't need access to `this` in their initialization outside of the constructor. we only use constructors when necessary, so adopting this as a convention makes it easier to add or remove a constructor because you don't have to move any prop declarations around.

Reply:
Not sure how I feel about this as a general convention, but sounds good for this set of work.